### PR TITLE
Add Address/Transaction search to home page SearchField

### PR DESCRIPTION
**Pull request by LuizAsFight**
Made the homepage search input functionally working.
Added address/transaction search, using existents graphql queries

image

### CODE REVIEW

File: src/pages/HomePage/index.tsx [hunk 1]
```diff
@@ -58,14 +78,38 @@ export default function HomePage() {
...
   };
 
+  const handleClickSearch = async () => {
+    if (isAllowedToSearch && searchText) {
+      const result = await searchTransactionQuery({
+        variables: { transaction: searchText, address: searchText },
+      });
+
+      if (result.data?.transaction?.id) {
+        window.location.assign(`/transaction/${searchText}`);
```
Comment (luizstacio) [inline src/pages/HomePage/index.tsx:88]: It's possible to use the router system here?

File: src/pages/HomePage/index.tsx [hunk 2]
```diff
@@ -58,14 +78,38 @@ export default function HomePage() {
...
+      } else if (result.data?.transactionsByOwner?.edges?.length) {
+        window.location.assign(`/address/${searchText}`);
+      } else {
+        setIsSearchNotFound(true);
+        setTimeout(() => setIsSearchNotFound(false), 1500);
```
Comment (luizstacio) [inline src/pages/HomePage/index.tsx:93]: Why we need this timeout?

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Comment by LuizAsFight** [comment]
Here's a demo of:

1- Search for a correct `transactionId`
2- Search for a correct `addressId`
3- Search for an invalid value

https://user-images.githubusercontent.com/8636507/168712565-156a5914-64fc-43b3-9b32-1da1371dbe2e.mov

File: src/pages/HomePage/index.tsx (hunk 2 above)
Comment (LuizAsFight) [reply src/pages/HomePage/index.tsx:94]: to show "Not found" message, then hide it after `1500 ms`

File: src/pages/HomePage/index.tsx (hunk 1 above)
Comment (LuizAsFight) [reply src/pages/HomePage/index.tsx:89]: I'll check

Comment (LuizAsFight) [reply src/pages/HomePage/index.tsx]: done.

File: src/pages/HomePage/index.tsx [hunk 3]
```diff
@@ -58,14 +80,39 @@ export default function HomePage() {
...
           <InputContainer>
-            <Input placeholder="Search an address, transaction or block" />
-            <SearchIcon />
+            <Input
+              placeholder="Search for transaction / address"
+              onChange={(e) => setSearchText(e?.target?.value)}
+            />
+            <SearchIcon
+              isDisabled={!isAllowedToSearch}
+              onClick={isAllowedToSearch ? handleClickSearch : undefined}
```
Comment (luizstacio) [inline src/pages/HomePage/index.tsx:113]: We should also search when clicking `enter` and also right after a `past`

**Review by luizstacio — APPROVED** [review]

**Pull request merged**